### PR TITLE
combine send and receive into a single integration

### PR DESCRIPTION
**Pull request by sammy-da**
single integration for the two send/receive components

### CODE REVIEW

File: dabl-meta.yaml [hunk 1]
```diff
@@ -13,17 +13,11 @@ catalog:
...
-      runtime: python-direct
-      fields: []
-    - id: com.projectdabl.integrations.slack_send_message
-      name: Slack Send Message
-      description: Allows outbound messages recorded on a ledger to be sent to Slack
-      entrypoint: core_int.integration_slack_send_message:integration_slack_send_main
-      env_class: core_int.integration_slack_send_message:IntegrationSlackSendMessageEnv
+    - id: com.projectdabl.integrations.slack_integration
+      name: Slack Integration
+      description: Allows outbound messages recorded on a ledger to be sent to Slack, and direct messages sent to a Slack application to be recorded on a ledger.
```
Comment (sammy-da) [inline dabl-meta.yaml]: copy/pasta 😄

File: src/core_int/integration_slack.py [hunk 2]
```diff
@@ -18,10 +18,29 @@
...
+        channel = event.cdata['slackChannel']
+        text = event.cdata['messageText']
+
+        sc.chat_postMessage(channel=channel, text=text)
+
+        return [exercise(event.cid, 'Archive')]
+
     @events.webhook.post(label='Slack Event Webhook Endpoint')
     async def on_webhook_post(request):
         body = await request.json()
```
Comment (sammy-da) [inline src/core_int/integration_slack.py:46]: `500` status if this fails, due to `hook_response` being `None`:
https://github.com/digital-asset/daml-dit-if/blob/master/daml_dit_if/main/integration_webhook_context.py#L48

File: dabl-meta.yaml (hunk 1 above)
Comment (mschaef-da) [reply dabl-meta.yaml]: "Allows messages to be sent to and received from Slack."

File: src/core_int/integration_slack.py (hunk 2 above)
Comment (mschaef-da) [reply src/core_int/integration_slack.py:46]: 500 status if parsing the request fails?

File: dabl-meta.yaml [hunk 3]
```diff
@@ -13,17 +13,11 @@ catalog:
...
-    - id: com.projectdabl.integrations.slack_send_message
-      name: Slack Send Message
-      description: Allows outbound messages recorded on a ledger to be sent to Slack
-      entrypoint: core_int.integration_slack_send_message:integration_slack_send_main
-      env_class: core_int.integration_slack_send_message:IntegrationSlackSendMessageEnv
+    - id: com.projectdabl.integrations.slack_integration
+      name: Slack Send and Receive Message
+      description: Allows messages to be sent to and received from Slack.
+      entrypoint: core_int.integration_slack:integration_slack_main
+      env_class: core_int.integration_slack:IntegrationSlackSendMessageEnv
```
Comment (mschaef-da) [inline dabl-meta.yaml]: Probably this class name should be `IntegrationSlackEnv` to align with the entry point `integration_slack_main`.

Comment (sammy-da) [reply dabl-meta.yaml]: my bad!

File: src/core_int/integration_slack.py (hunk 2 above)
Comment (sammy-da) [reply src/core_int/integration_slack.py:46]: yes.
```bash
curl -X POST https://integrations.projectdabl.com/integration/ijo8doscmuqkyicz
500 Internal Server Error
```

Comment (mschaef-da) [reply src/core_int/integration_slack.py:46]: I think that's probably acceptable, although we do need to do some overall robustness work on this integration as part of a later PR.

**Comment by mschaef-da** [comment]
This looks good... how have you tested it?

File: src/core_int/integration_slack.py (hunk 2 above)
Comment (sammy-da) [reply src/core_int/integration_slack.py:46]: yep my thoughts as well.

**Comment by sammy-da** [comment]
> This looks good... how have you tested it?

```
curl -X POST https://integrations.sammyabed.projectdabl-dev.com/integration/qk5fu1rzesnwvyry -d '{"type": "event_callback", "event": {"type": "message", "subtype": "blah", "channel": "mychan", "user": "samtheman", "text": "hello"}}'
200: OK```

with a `SlackIntegration.InboundDirectMessage:InboundDirectMessage` with correct payload created on the ledger!

**Comment by sammy-da** [comment]
> This looks good... how have you tested it?

```
curl -X POST https://integrations.projectdabl.com/integration/qk5fu1rzesnwvyry -d '{"type": "event_callback", "event": {"type": "message", "subtype": "blah", "channel": "mychan", "user": "samtheman", "text": "hello"}}'
200: OK
```
A `SlackIntegration.InboundDirectMessage:InboundDirectMessage` with correct payload created on the ledger!

**Comment by sammy-da** [comment]
And just tested the Send functionality.
Message sent successfully to a test slack channel!

```
2021-02-23T15:45:12+0000 [DEBUG] (slack.web.slack_response) Received the following response - status: 200, headers: {'Date': 'Tue, 23 Feb 2021 15:45:12 GMT', 'Server': 'Apache', 'x-xss-protection': '0', 'Pragma': 'no-cache', 'Cache-Control': 'private, no-cache, no-store, must-revalidate', 'Access-Control-Allow-Origin': '*', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-slack-req-id': '224f2db89aa2f32f076e6b5a0a866566', 'x-content-type-options': 'nosniff', 'referrer-policy': 'no-referrer', 'Access-Control-Expose-Headers': 'x-slack-req-id, retry-after', 'x-slack-backend': 'r', 'x-oauth-scopes': 'incoming-webhook,chat:write', 'x-accepted-oauth-scopes': 'chat:write', 'Expires': 'Mon, 26 Jul 1997 05:00:00 GMT', 'Vary': 'Accept-Encoding', 'Access-Control-Allow-Headers': 'slack-route, x-slack-version-ts, x-b3-traceid, x-b3-spanid, x-b3-parentspanid, x-b3-sampled, x-b3-flags', 'Content-Encoding': 'gzip', 'Content-Length': '315', 'Content-Type': 'application/json; charset=utf-8', 'x-envoy-upstream-service-time': '66', 'x-backend': 'main_normal main_canary_with_overflow main_control_with_overflow', 'x-server': 'slack-www-hhvm-main-iad-3676', 'x-via': 'envoy-www-iad-nomv, haproxy-edge-iad-dm1q', 'x-slack-shared-secret-outcome': 'shared-secret', 'Via': 'envoy-www-iad-nomv'}, body: {'ok': True, 'channel': 'C01P1KFUCHY', 'ts': '1614095112.000900', 'message': {'bot_id': 'B01P1UW8S66', 'type': 'message', 'text': 'hello', 'user': 'U01NV62P2KG', 'ts': '1614095112.000900', 'team': 'T01P7Q2K620', 'bot_profile': {'id': 'B01P1UW8S66', 'deleted': False, 'name': 'my-test-app', 'updated': 1614094628, 'app_id': 'A01P7QLKFT6', 'icons': {'image_36': 'https://a.slack-edge.com/80588/img/plugins/app/bot_36.png', 'image_48': 'https://a.slack-edge.com/80588/img/plugins/app/bot_48.png', 'image_72': 'https://a.slack-edge.com/80588/img/plugins/app/service_72.png'}, 'team_id': 'T01P7Q2K620'}}}

**Review by mschaef-da — APPROVED** [review]
LGTM

**Pull request merged**